lib/inspec/dependencies/dependency_set.rb

```diff
@@ -18,14 +18,22 @@ class DependencySet
     # @param cwd [String] Current working directory for relative path includes
     # @param vendor_path [String] Path to the vendor directory
     #
-    def self.from_lockfile(lockfile, cwd, vendor_path)
+    def self.from_lockfile(lockfile, cwd, vendor_path, backend)
       vendor_index = VendorIndex.new(vendor_path)
       dep_tree = lockfile.deps.map do |dep|
-        Inspec::Requirement.from_lock_entry(dep, cwd, vendor_index)
+        Inspec::Requirement.from_lock_entry(dep, cwd, vendor_index, backend)
```

Comment:
Somehow it feels strange that the dependency resolution needs a backend? Why is that required?